messaging/kafka/src/main/java/io/helidon/messaging/kafka/SimpleKafkaConsumer.java

```diff
@@ -0,0 +1,322 @@
+     * @param config      Helidon {@link io.helidon.config.Config config}
+     * @see KafkaConfigProperties
+     * @see io.helidon.config.Config
+     */
+    public SimpleKafkaConsumer(String channelName, Config config) {
+        this(channelName, config, null);
+    }
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param channelName     key in configuration
+     * @param config          Helidon {@link io.helidon.config.Config config}
+     * @param consumerGroupId Custom group.id, can be null, overrides group.id from configuration
+     * @see KafkaConfigProperties
+     * @see io.helidon.config.Config
+     */
+    public SimpleKafkaConsumer(String channelName, Config config, String consumerGroupId) {
+        this.properties = new KafkaConfigProperties(config.get("mp.messaging.incoming").get(channelName));
+        this.properties.setProperty(KafkaConfigProperties.GROUP_ID, getOrGenerateGroupId(consumerGroupId));
+        this.topicNameList = properties.getTopicNameList();
+        this.consumerId = channelName;
+        this.consumer = new KafkaConsumer<>(properties);
+    }
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     */
+    public SimpleKafkaConsumer(Config config) {
+        this.properties = new KafkaConfigProperties(config);
+        this.properties.setProperty(KafkaConfigProperties.GROUP_ID, getOrGenerateGroupId(null));
+        this.topicNameList = properties.getTopicNameList();
+        this.consumerId = null;
+        this.consumer = new KafkaConsumer<>(properties);
+    }
+
+    /**
+     * Execute supplied consumer for each received record.
+     *
+     * @param function to be executed for each received record
+     * @return {@link java.util.concurrent.Future}
+     */
+    public Future<?> consumeAsync(Consumer<ConsumerRecord<K, V>> function) {
+        return this.consumeAsync(Executors.newWorkStealingPool(), null, function);
+    }
+
+    /**
+     * Execute supplied consumer by provided executor service for each received record.
+     *
+     * @param executorService Custom executor service used for spinning up polling thread and record consuming threads
+     * @param customTopics    Can be null, list of topics appended to the list from configuration
+     * @param function        Consumer method executed in new thread for each received record
+     * @return The Future's get method will return null when consumer is closed
+     */
+    public Future<?> consumeAsync(ExecutorService executorService, List<String> customTopics,
+                                  Consumer<ConsumerRecord<K, V>> function) {
+        LOGGER.info(String.format("Initiating kafka consumer %s listening to topics: %s with groupId: %s",
```

Comment:
Info messages should be only printed if we know it will happen only once per runtime of Helidon. I am not sure this is the case.

Reply:
This is removed too.